docs/modules/ROOT/pages/index.adoc

```diff
@@ -43,10 +43,236 @@ Once the BOM is added, simply add the `io.quarkiverse.artemis:quarkus-artemis-jm
+[source,java,subs=attributes+]
+----
+public class MyBeanProvider {
+    @ApplicationScoped
+    @Identified("my-connection-factory") // this annotation makes it so the bean is picked up by the health endpoint
+    public ConnectionFactory myConnectionFactory(...) {
+        ConnectionFactory myConnectionFactory = ...;
+        ...
+        return myConnectionFactory;
+    }
+}
+----
+
+If we do not want that beans created within our application is picked up by the health endpoint, we can set `quarkus.artemis.health.external.enabled` to *false*.
+
+IMPORTANT: Note that `ServerLocator` s / `ConnectionFactory` s are only picked up when `quarkus.artemis.health.enabled` is *true*.
 
 == Artemis DevServices
-Artemis DevServices are automatically enabled unless `quarkus.artemis.url` is set or `quarkus.artemis.devservices.enabled` is *false* explicitly. And if you still want to use `ArtemisTestResource` in the test, you need to disable artemis devservices.
+Artemis DevServices are automatically enabled unless `quarkus.artemis.url` / `quarkus.artemis."named-configuration".url` is set or `quarkus.artemis.devservices.enabled` / `quarkus.artemis."named-configuration".enabled` is *false* explicitly. And if you still want to use `ArtemisTestResource` in the test, you need to disable artemis devservices.
+
+== Test framework
+To facilitate testing, we provide a dedicated testing framework that can start an in-memory artemis instance  For the default configuration, we can start an in-memory instance of artemis by annotating our tests with `@QuarkusTestResource(ArtemisTestResource.class)`. The embedded artemis can be configured through a `broker.xml`, located under `src/test/resources`.
+
+To start an in-memory artemis for a named configuration, we can define our own test annotation by extending `ArtemisTestResource.class`:
+
+.Annotation to start an in-memory artemis instance for the named configuration "named-1"
+[source,java,subs=attributes+]
+----
+public class NamedOneArtemisTestResource extends ArtemisTestResource {
+    public NamedOneArtemisTestResource() {
+        super("named-1");
+    }
+}
+----
+
+We can then annotate our test with `@QuarkusTestResource(NamedOneArtemisTestResource.class)`.
+
+This instance can be configured by placing a `broker-named-1.xml` in `src/test/resources`. In general, for a configuration with name `<name>`, we can configure the corresponding in-memory artemis through file `src/test/resources/broker-<name>.xml`.
+
+For an in-depth explanation of what can be configured in a `broker.xml`, please see the link:https://activemq.apache.org/components/artemis/documentation/latest/configuration-index.html:[official Artemis documentation].
+
+== Examples
+Examples can be found in the `integration-tests` module, e.g link:https://github.com/quarkiverse/quarkus-artemis/tree/main/integration-tests/core/with-default/src/test/resources[The Apache ActiveMQ Artemis Core integration with default configuration tests module]
```

Comment:
@middagj  @zhfeng 
I'd like to have a review of this part in particular. I am sure that there must be a better way to link to a project resource.

Reply:
I think a relative reference should work: https://github.blog/2013-01-31-relative-links-in-markup-files/

Reply:
hm so...  `file:///../../../../integration-tests/core/with-default/src/test[The Apache ActiveMQ Artemis Core integration with default configuration tests module]` should work?

Reply:
I thought more like:

```adoc
file://integration-tests/core/with-default/src/test[The Apache ActiveMQ Artemis Core integration with default configuration tests module]
```

Problem here is the feedback loop of 6 hours?

Reply:
... yes. Let's embrace agile: move fast, break things :slightly_smiling_face: